tsl/src/nodes/decompress_chunk/decompress_chunk.c

```diff
@@ -544,6 +544,132 @@ compressed_rel_setup_reltarget(RelOptInfo *compressed_rel, CompressionInfo *info
+		memcpy(newinfo, oldinfo, sizeof(RestrictInfo));
+
+		/* Recursively fix the clause itself */
+		newinfo->clause = (Expr *) chunk_joininfo_mutator((Node *) oldinfo->clause, context);
+
+		/* and the modified version, if an OR clause */
+		newinfo->orclause = (Expr *) chunk_joininfo_mutator((Node *) oldinfo->orclause, context);
+
+		/* adjust relid sets too */
+		newinfo->clause_relids =
+			decompress_chunk_adjust_child_relids(oldinfo->clause_relids,
+												 context->chunk_rel->relid,
+												 context->compressed_rel->relid);
+		newinfo->required_relids =
+			decompress_chunk_adjust_child_relids(oldinfo->required_relids,
+												 context->chunk_rel->relid,
+												 context->compressed_rel->relid);
+		newinfo->outer_relids =
+			decompress_chunk_adjust_child_relids(oldinfo->outer_relids,
+												 context->chunk_rel->relid,
+												 context->compressed_rel->relid);
+		newinfo->nullable_relids =
+			decompress_chunk_adjust_child_relids(oldinfo->nullable_relids,
+												 context->chunk_rel->relid,
+												 context->compressed_rel->relid);
+		newinfo->left_relids = decompress_chunk_adjust_child_relids(oldinfo->left_relids,
+																	context->chunk_rel->relid,
+																	context->compressed_rel->relid);
+		newinfo->right_relids =
+			decompress_chunk_adjust_child_relids(oldinfo->right_relids,
+												 context->chunk_rel->relid,
+												 context->compressed_rel->relid);
+
+		newinfo->eval_cost.startup = -1;
+		newinfo->norm_selec = -1;
+		newinfo->outer_selec = -1;
+		newinfo->left_em = NULL;
+		newinfo->right_em = NULL;
+		newinfo->scansel_cache = NIL;
+		newinfo->left_bucketsize = -1;
+		newinfo->right_bucketsize = -1;
+#if PG11_GE
+		newinfo->left_mcvfreq = -1;
+		newinfo->right_mcvfreq = -1;
+#endif
+		return (Node *) newinfo;
+	}
+	return expression_tree_mutator(node, chunk_joininfo_mutator, context);
+}
+
+/* translate chunk_rel->joininfo for compressed_rel
+ * this is necessary for create_index_path which gets join clauses from
+ * rel->joininfo and sets up paramaterized paths (in rel->ppilist).
+ * ppi_clauses is finally used to add any additional filters on the
+ * indexpath when creating a plan in create_indexscan_plan.
+ * Otherwise we miss additional filters that need to be applied after
+ * the index plan is executed (github issue 1558)
+ */
+static void
+compressed_rel_setup_joininfo(RelOptInfo *compressed_rel, CompressionInfo *info)
+{
+	RelOptInfo *chunk_rel = info->chunk_rel;
+	ListCell *lc;
+	List *compress_joininfo = NIL;
+	foreach (lc, chunk_rel->joininfo)
+	{
+		RestrictInfo *compress_ri;
+		RestrictInfo *ri = (RestrictInfo *) lfirst(lc);
+		// RestrictInfo *compress_ri = expression_tree_mutator((Node *) ri, chunk_joininfo_mutator,
+		// (void*)info);
+		Node *result = chunk_joininfo_mutator((Node *) ri, info);
+		Assert(IsA(result, RestrictInfo));
+		compress_ri = (RestrictInfo *) result;
+		// Assert(compress_ri->is_pushed_down == false); fails for exp.txt
```

Comment:
why is this comment about?

Reply:
this is a comment from when I didn't understand what rels have is_pushed_down set. removed assert and comment.